src/Testing/Asserts.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+namespace SergiX44\Nutgram\Testing;
+
+use GuzzleHttp\Psr7\Request;
+use Illuminate\Support\Arr;
+use Illuminate\Testing\Assert as LaraUnit;
+use InvalidArgumentException;
+use JsonException;
+use PHPUnit\Framework\Assert as PHPUnit;
+
+/**
+ * @mixin FakeNutgram
+ */
+trait Asserts
+{
+    /**
+     * @param  callable  $closure
+     * @param  int  $index
+     * @return $this
+     */
+    public function assertRaw(callable $closure, int $index = 0): self
+    {
+        $reqRes = $this->testingHistory[$index];
+
+        /** @var Request $request */
+        [$request,] = array_values($reqRes);
+
+        PHPUnit::assertTrue($closure($request));
+
+        return $this;
+    }
+
+    /**
+     * @param  string  $method
+     * @param  int  $times
+     * @return $this
+     */
+    public function assertCalled(string $method, int $times = 1): self
+    {
+        $actual = 0;
+        foreach ($this->testingHistory as $reqRes) {
+            /** @var Request $request */
+            [$request,] = array_values($reqRes);
+
+            if ($request->getUri()->getPath() === $method) {
+                $actual++;
+            }
+        }
+
+        PHPUnit::assertEquals($times, $actual, "The method '$method' was called $actual instead of $times");
+
+        return $this;
+    }
+
+    /**
+     * @param  string|string[]  $method
+     * @param  array|null  $expected
+     * @param  int  $index
+     * @return $this
+     */
+    public function assertReply(string|array $method, ?array $expected = null, int $index = 0): self
+    {
+        $reqRes = $this->testingHistory[$index];
+
+        /** @var Request $request */
+        [$request,] = array_values($reqRes);
+
+        PHPUnit::assertContains($request->getUri()->getPath(), Arr::wrap($method), 'Method name not found');
+
+        if ($expected !== null) {
+            try {
+                $expected = json_decode(json_encode($expected), true);
+                $actual = json_decode((string)$request->getBody(), true, flags: JSON_THROW_ON_ERROR);
+            } catch (JsonException) {
+                $actual = [];
+            }
+
+            LaraUnit::assertArraySubset($expected, $actual, msg: 'Sub array not found in the request body');
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param  array  $expected
+     * @param  int  $index
+     * @param  string|null  $forceMethod
+     * @return $this
+     */
+    public function assertMessage(array $expected, int $index = 0, ?string $forceMethod = null): self
+    {
+        $allowed = $forceMethod ?? [
```

Comment:
This list must be mantained, consider retrieving the methods on instantiation via reflection instead